test/tests/kernels/hfem/dirichlet.i

```diff
@@ -0,0 +1,99 @@
+[Mesh]
+  [square]
+    type = GeneratedMeshGenerator
+    nx = 3
+    ny = 3
+    dim = 2
+  []
+  build_all_side_lowerd_mesh = true
+[]
+
+[Variables]
+  [u]
+    order = THIRD
+    family = MONOMIAL
+    block = 0
+  []
+  [lambda]
+    order = CONSTANT
+    family = MONOMIAL
+    block = INTERNAL_SIDE_LOWERD_SUBDOMAIN
+  []
+  [lambdab]
```

Comment:
Looks like we do not have any kernels for `lambdab` so we are not really testing the other way of imposing Dirichlet BCs. I suggest to either add the kernels for `lambdab` or remove this variable completely.

Reply:
`lambdab` is used in here:
```java
[BCs]
  [all]
    type = HFEMDirichletBC
    boundary = 'left right top bottom'
    variable = u
    lowerd_variable = lambdab
  []
[]
```
Do you expect me to delete this one?

Reply:
instead of a lowerd_variable here, we can directly use the given value. Essentially, lambdab is a primal variable, but no equation is for it currently.

Reply:
I added additional kernels to `lambdab`: 
```
  [reaction]
    type = Reaction
    variable = uhat
    rate = '1'
    block = BOUNDARY_SIDE_LOWERD_SUBDOMAIN
    reaction_coefficient = rc
  []
  [uhat_coupled]
    type = CoupledForce
    variable = uhat
    block = BOUNDARY_SIDE_LOWERD_SUBDOMAIN
    v = lambdab
    coef = '1'
  []
```